print-ospf.c

```diff
@@ -611,6 +654,226 @@ ospf_print_tos_metrics(netdissect_options *ndo,
     }
 }
 
+static int
+ospf_print_ri_lsa_sid_label_range_tlv(netdissect_options *ndo, const uint8_t *tptr, u_int tlv_length)
+{
+    u_int subtlv_type, subtlv_length;
+
+    while (tlv_length >= 4) {
+
+	ND_TCHECK_4(tptr);
+	subtlv_type = GET_BE_U_2(tptr);
+	subtlv_length = GET_BE_U_2(tptr+2);
+	tptr+=4;
+	tlv_length-=4;
+
+	/* Infinite loop protection. */
+	if (subtlv_type == 0 || subtlv_length == 0) {
+	    return -1;
+	}
+
+	ND_PRINT("\n\t      %s subTLV (%u), length: %u, value: ",
+		 tok2str(lsa_opaque_ri_sid_subtlv_values,"unknown",subtlv_type),
+		 subtlv_type,
+		 subtlv_length);
+
+	switch (subtlv_type) {
+	case LS_OPAQUE_RI_SUBTLV_SID_LABEL:
+	    if (subtlv_length == 3) {
+		ND_PRINT("\n\t\tLabel: %u", GET_BE_U_3(tptr));
+	    } else if (subtlv_length == 4) {
+		ND_PRINT("\n\t\tSID: %u", GET_BE_U_4(tptr));
+	    } else {
+		ND_PRINT("\n\t\tBogus subTLV length %u", subtlv_length);
+	    }
+	    break;
+
+	default:
+	    if (ndo->ndo_vflag <= 1) {
+		if (!print_unknown_data(ndo, tptr, "\n\t\t", subtlv_length))
+		    return -1;
+	    }
+	}
+
+	/* in OSPF everything has to be 32-bit aligned, including subTLVs */
+	if (subtlv_length % 4) {
+	    subtlv_length += (4 - (subtlv_length % 4));
+	}
+	tptr+=subtlv_length;
+	tlv_length-=subtlv_length;
+    }
+    return 0;
+trunc:
+    return -1;
+}
+
+static int
+ospf_print_ep_lsa_extd_prefix_tlv(netdissect_options *ndo, const uint8_t *tptr, u_int tlv_length)
+{
+    u_int subtlv_type, subtlv_length;
+    uint8_t flags, mt_id, algo;
+
+    while (tlv_length >= 4) {
+	ND_TCHECK_4(tptr);
+	subtlv_type = GET_BE_U_2(tptr);
+	subtlv_length = GET_BE_U_2(tptr+2);
+	tptr+=4;
+	tlv_length-=4;
+
+	/* Infinite loop protection. */
+	if (subtlv_type == 0 || subtlv_length == 0) {
```

Comment:
nd_print_invalid(ndo); before the return?

Reply:
ok